x-pack/test/security_solution_api_integration/test_suites/detections_response/detection_engine/exceptions/workflows/basic_license_essentials_tier/create_rule_exceptions_ess.ts

```diff
@@ -79,25 +78,15 @@ export default ({ getService }: FtrProviderContext) => {
           })
           .expect(200);
 
-        /**
-         * Confirm type on SO so that it's clear in the tests whether it's expected that
-         * the SO itself is migrated to the inteded object type, or if the transformation is
-         * happening just on the response. In this case, change will
-         * NOT include a migration on SO.
-         */
         const {
           hits: {
             hits: [{ _source: ruleSO }],
           },
         } = await getRuleSOById(es, ruleWithLegacyInvestigationField.id);
-        const isInvestigationFieldMigratedInSo = await checkInvestigationFieldSoValue(ruleSO, {
```

Comment:
Now, the legacy investigation field migration happens much earlier. With this refactoring, we convert data to our internal representation `RuleResponse` immediately after reading a rule from `RulesClient`. This means that instead of transforming the data only before returning it in the response, we transform it as soon as we read it.

This approach ensures that in our API handlers, the data is always in the new format and thus gets migrated on any write or update. This is much more straightforward than migrating the data only on certain writes as before.